design-documents/graph-ql/coverage/b2b/purchase-order.graphqls

```diff
@@ -0,0 +1,303 @@
+    shipping_address: CustomerAddress @doc(description: "The shipping address for the purchase order")
+    billing_address: CustomerAddress @doc(description: "The billing address for the purchase order")
+    carrier: String @doc(description: "The shipping carrier for the purchase order delivery")
+    shipping_method: String @doc(description: "The delivery method for the purchase order")
+    items(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderItems @doc(description: "Items that belong to the purchase order")
+    history_log(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderHistoryLog @doc(description: "The log of the events related to the purchase order")
+    approval_flow: PurchaseOrderApprovalFlow @doc(description: "The log of the events related to the purchase order approval flow")
+    available_actions: [PurchaseOrderAction] @doc(description: "Purcahse order actions available to the customer. Can be used to display action buttons on the client")
+}
+
+enum PurchaseOrderAction {
+    REJECT
+    CANCEL
+    APPROVE
+}
+
+type PurchaseOrderApprovalFlow {
+    items: [PurchaseOrderApprovalFlowItem]!
+}
+
+type PurchaseOrderApprovalFlowItem {
+    uid: ID! @doc(description: "Unique identifier of the purchase order flow item.")
+    title: String! @doc(description: "Summary of the event related to purchase order approval flow")
+    description: String! @doc(description: "Description of the event related to purchase order approval flow")
+    status: PurchaseOrderApprovalFlowItemStatus! @doc(description: "Status associated with the event related to purchase order approval flow")
+}
+
+enum PurchaseOrderApprovalFlowItemStatus {
+    PENDING
+    APPROVED
+    REJECTED
+}
+
+type PurchaseOrderComments {
+    items: [PurchaseOrderComment]!
+    page_info: SearchResultPageInfo
+    total_count: Int!
+}
+
+type PurchaseOrderComment {
+    uid: ID! @doc(description: "Unique identifier of the comment.")
+    timestamp: String! @doc(description: "The date and time when the comment was created")
+    author: PurchaseOrderCommentAuthor! @doc(description: "The name of the user who left the comment")
+    text: String! @doc(description: "The text of the comment")
+}
+
+type PurchaseOrderCommentAuthor {
+    firstname: String! @doc(description: "First name of the user who left the purchase order comment")
+    lastname: String! @doc(description: "Last name of the user who left the purchase order comment")
+}
+
+type PurchaseOrderHistoryLog {
+    items: [PurchaseOrderHistoryItem]!
+    page_info: SearchResultPageInfo
+    total_count: Int
+}
+
+type PurchaseOrderHistoryItem {
+    uid: ID! @doc(description: "Unique identifier of the purchase rder history item.")
+    timestamp: String! @doc(description: "The date and time when the event happened.")
+    description: String! @doc(description: "Description of the event.")
+}
+
+type PurchaseOrderItems {
```

Comment:
Not related to this PR, but any reasons of why we shouldn't implement the same for `CustomerOrder`?

Reply:
Don't know of any reason, can consider it.